node-puppeteer-chrome/puppeteer_utils.js

```diff
@@ -0,0 +1,100 @@
+const { version: puppeteerVersion } = require('puppeteer/package.json');
+
+const VERSION_REGEX = /([\d.?]+)/g;
+const chromeVersionDownloadUrl = version => `https://dl.google.com/linux/chrome/deb/pool/main/g/google-chrome-stable/google-chrome-stable_${version}_amd64.deb`;
+
+/**
+ * @param {string} text
+ * @returns {{ chrome: string; pptr: string; }[]}
+ */
+function parseCompatibilityVersions(text) {
+    const versionsText = text.substring(
+        text.indexOf('Releases per Chromium Version:'),
+        text.indexOf('* [All releases]')
+    );
+    const versions = versionsText.split('\n')
+        .filter(line => line.includes('* Chromium'))
+        .map(line => {
+            const vers = line.match(VERSION_REGEX);
+            return { chrome: vers[0], pptr: vers[1] }
+        });
+    if (versions.length < 5){
+        throw new Error('Cannot find and parse Puppeteer-Chromium versions from Markdown file.');
+    }
+    return versions;
+}
+
+/**
+ * @param {string} compatible
+ * @param {string} actual
+ * @returns {boolean}
+ */
+function areVersionsCompatible(compatible, actual) {
+    const [compMajor, compMinor] = compatible.split('.').map(n => Number(n));
+    const [actualMajor, actualMinor] = actual.split('.').map(n => Number(n));
+    return actualMajor === compMajor && actualMinor >= compMinor;
+}
+
+/**
+ * @param {import('puppeteer').Browser} browser
+ * @returns
+ */
+async function fetchCompatibilityVersions(browser) {
+	const page = await browser.newPage();
+    const apiResponse = await page.goto('https://raw.githubusercontent.com/GoogleChrome/puppeteer/main/docs/api.md');
+    const compatibilityVersions = parseCompatibilityVersions(await apiResponse.text());
+
+	return compatibilityVersions;
+}
+
+/**
+ * Downloads the closest version of Chrome available
+ * @param {import('puppeteer').Browser} browser
+ * @param {string} versionToCheck
+ */
+async function downloadClosestChromeInstaller(browser, versionToCheck) {
+	// Get the first 3 parts of the version
+	const relevantPart = versionToCheck.split('.').slice(0, 3).join('.');
+	const page = await browser.newPage();
+	// Lazy load node-fetch here, since it won't be present outside the download process
+	const fetch = require('node-fetch');
```

Comment:
Why not require at top? The performance gains from not requiring are negligible.

Reply:
This file is shared between the tester and the downloader - the former of which won't have node-fetch present, thus causing an error to be thrown